threads.go

```diff
@@ -708,8 +707,12 @@ func (t *threads) startPulling() {
 		}
 	}
 	timer := time.NewTimer(InitialPullInterval)
-	<-timer.C
-	pull()
+	select {
+	case <-timer.C:
+		pull()
+	case <-t.ctx.Done():
+		return
+	}
```

Comment:
This bug took me quite a bit to find. (wasn't related with any created fix).

The problem here was that if the client closes the `Threadservice` before `InitialPullInterval`, the `Close()` method of `Threadservice` will close the datastore, and will cause `pull()` internal work to panic strangely at the badger level (since the database was closed!)

Reply:
Ooo, sneaky one!